src/app/components/quick-pick/quick-pick.component.ts

```diff
@@ -0,0 +1,163 @@
+             *ngIf="formGroup.controls['expressionField']">
+            
+            <expression-input class="col-sm-11 expression-input"
+                      [control]="formGroup.controls['expressionField']"
+                      [isExpression]="!!expressionModel.serialize().script"
+                      (onEdit)="editExpression()"
+                      (onClear)="clearExpression()">
+            </expression-input>
+            
+            <span class="col-sm-1">
+                <i class="fa fa-trash clear-icon" (click)="removeExpressionInput()"></i>
+            </span>
+        </div>
+    `
+})
+export class QuickPickComponent {
+
+    @Input()
+    public buttonList: String[];
+
+    @Input()
+    public context: any;
+
+    @Input()
+    private formGroup: FormGroup;
+
+    @Output()
+    public onUpdate = new ReplaySubject<string>(1);
+
+    private expressionModel: ExpressionModel;
+
+    private expressionInputSub: Subscription;
+
+    private subs: Subscription[] = [];
+
+    constructor(private expressionSidebarService: ExpressionSidebarService) {
+
+    }
+
+    ngOnInit() {
+        this.formGroup.addControl(
+            "radioButtonControl", new FormControl("")
+        );
+
+        this.subs.push(
+            this.formGroup.controls["radioButtonControl"].valueChanges.subscribe((value: string) => {
+                this.onUpdate.next(value);
+            })
+        );
+    }
+
+    private crateExpressionInput(): void {
+        if (this.formGroup.contains("expressionField")) {
+            return;
+        }
+
+        this.formGroup.controls["radioButtonControl"].setValue("");
+        this.expressionModel = new ExpressionModel("");
+
+        this.formGroup.addControl(
+            "expressionField",
+            new FormControl(this.expressionModel.serialize(), Validators.compose([Validators.required, Validators.minLength(1)]))
+        );
+
+        this.subs.push(
+            this.formGroup.controls["expressionField"].valueChanges
+                .debounceTime(300)
+                .subscribe((value: string) => {
+                    this.updateExpressionValue(new ExpressionModel(value));
+                })
+        );
+    }
+
+    private removeExpressionInput(): void {
+        this.clearExpression();
+        this.formGroup.removeControl("expressionField");
+    }
+
+    private editExpression(): void {
+        const newExpression: BehaviorSubject<ExpressionModel> = new BehaviorSubject<ExpressionModel>(undefined);
+        this.removeExpressionInputSub();
+
+        this.expressionInputSub = newExpression
+            .filter(expression => expression !== undefined)
+            .subscribe((expression: ExpressionModel) => {
+                this.updateExpressionValue(expression);
+
+                this.formGroup.controls["expressionField"].setValue(
+                    this.expressionModel.getExpressionScript(), {
+                        onlySelf: true,
```

Comment:
On line 106, I listen to the value change, inside the input field, which sets the new value to a string. I don't want to trigger that change detection if the the value was set to an expression, but I do want to update the input field's value.